Source/DataSources/EntityCollection.js

```diff
@@ -47,8 +47,11 @@ define([
      * An observable collection of {@link Entity} instances where each entity has a unique id.
      * @alias EntityCollection
      * @constructor
+     *
+     * @param {DataSource|CompositeEntityCollection} dataSource The data source which created this collection.
      */
-    var EntityCollection = function() {
+    var EntityCollection = function(dataSource) {
+        this.dataSource = dataSource;
```

Comment:
1. I think `owner` would be a better name here instead of `dataSource`, since the owner can be a `CompositeEntityCollection` or `DataSource` instance.
2. It needs to be marked optional in the documentation and the description updated to reflect the possibility of a `CompositeEntityCollection`
3. I would defined `this._owner` here and then add a readonly getter property to ensure it's not modified after construction.